online/src/main/scala/ai/chronon/online/SparkInternalRowConversions.scala

```diff
@@ -116,7 +116,7 @@ object SparkInternalRowConversions {
   // The purpose of this class is to be used on fetcher output in a fetching context
   // we take a data type and build a function that operates on actual value
   // we want to build functions where we only branch at construction time, but not at function execution time.
-  def to(dataType: types.DataType, structToMap: Boolean = true): Any => Any = {
+  def to(dataType: types.DataType, structToMap: Boolean): Any => Any = {
```

Comment:
do we need to mark the internal row as structToMap = false?

Reply:
@yuli-han WDYM?

Reply:
so I ended up not hard coding structToMap in this function, but passing the structToMap value down. I think that's more intuitive. 

Instead, the outside caller of `to` will specify structToMap, so:
- inputEncoder will set it to true
- inputArrEncoder will set it to false